### PR TITLE
added some more integration tests for catalog, briefs

### PR DESCRIPTION
**Pull request by matt-sm**
Created tests for basic brief flow and catalog search.

Still uncertain where this should be integrated in CI process.

### CODE REVIEW

File: tests/integration/features/test_build_brief.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+@when('I enter a title')
+def enter_title(title, browser):
+    browser.fill('title', title)
+
+
+@when(parsers.parse('I enter {text} into {input}'))
+def enter_text(text, input, browser):
+    for t, i in zip(text.split(), input.split()):
+        if (t == 'text'):
+            t = random_string()
```
Comment (sarneaud) [inline tests/integration/features/test_build_brief.py]: These random strings are just lowercase letters of a fixed 10 characters (no spaces or punctuation, etc) so I'm not sure the randomness helps make the tests more powerful than just using a constant "lorem" everywhere.  On the other hand, if anything, if the randomness uncovers a new failure, it won't be reproducible unless you set a constant seed on each test case.

Is there anything wrong with tests reading like "I enter My Company Name into organisation"?

File: tests/integration/features/test_build_brief.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+@when('I enter a title')
+def enter_title(title, browser):
+    browser.fill('title', title)
+
+
+@when(parsers.parse('I enter {text} into {input}'))
+def enter_text(text, input, browser):
+    for t, i in zip(text.split(), input.split()):
```
Comment (sarneaud) [inline tests/integration/features/test_build_brief.py]: This zip/split trick only helps when you have a list of single-word values to put into a list of inputs.  In other scenarios, like putting a multi-word value into one input, it causes possibly surprising failures, or just gets in the way.  Is it really pulling its weight as a feature?

File: tests/integration/features/test_build_brief.py (hunk 1 above)
Comment (matt-sm) [inline tests/integration/features/test_build_brief.py]: Agreed - I originally created the random string for the brief name which will need to be unique, but everything else can have prescribed text.

File: tests/integration/features/test_build_brief.py (hunk 2 above)
Comment (matt-sm) [inline tests/integration/features/test_build_brief.py]: The zip/split makes this case work: 
`And I enter 30 20 50 into technicalWeighting culturalWeighting priceWeighting`
and waiting to do `save_and_continue` until all inputs are filled.  I could split on a ',' to make it safer so this multi-input case still works?

File: tests/integration/features/test_build_brief.py (hunk 1 above)
Comment (sarneaud) [inline tests/integration/features/test_build_brief.py]: I see.  Maybe we can add a dummy brief to the code that never really gets published.  (I.e., the code pretends to publish it, but doesn't actually do it, so it's only used for testing.)  We only need to test the flows here, not the underlying functionality.  Of course, that would be an improvement for another PR.

File: tests/integration/features/test_build_brief.py (hunk 2 above)
Comment (sarneaud) [inline tests/integration/features/test_build_brief.py]: Yeah, I see it's the save_and_continue thing.  Sure, splitting on ", " would help for now.  Maybe we can add a smarter parser later (the BDD framework has some more advanced options).

**Comment by matt-sm** [comment]
I removed the random string on the inputs and specify the actual text.  Future PR to create a 'test brief' so that we don't keep creating new briefs when running tests.
